### DISCUSSION TITLE
Ublock freezes MS Edge after a while when using search engines

### DISCUSSION THREAD

**Discussion by Josh65-2201** [uBlockOrigin/uBlock-issues#2769] [Q&A] [answered]
So I have elements blocked on Google and DuckDuckGo search engines with two issues all other site aren't having them.
DuckDuckGo will have an issue were after 6 hours of Edge being open it will get slow and slow with Edge being completely unresponsive after 2 more hours. 
Google has an issue were if it open a link in a new tab then change the page/image on google the tab just closes this also happens of Firefox.

**Comment by gorhill** [comment]
How did you come to the conclusion that uBO is the issue?

**Reply by Josh65-2201** [reply]
It only happens with Ublock enabled in Edge extensions menu and only started after Ublock was installed

**Reply by gorhill** [reply]
I won't be able to do much on my side really. When this happens, first bring up the browser task manager to find out which process is having undue CPU usage. If it's a specific process, use the performance profiler in this process to find out what is causing the undue CPU usage. Since it's not something I have seen or heard of before, at this point I am skeptical this is a uBO issue -- you will have to make the case this is an issue with uBO. If your whole browser freezes, it's unlikely a uBO issue since uBO execute in its own process.

---

> I have elements blocked on Google and DuckDuckGo search engines

Note that you mention having custom filters, please disclose these.

**Reply by Josh65-2201** [reply]
I will check the performance profile when it happens again. As for the browser task manger the CPU spike is on browser process only.
Here are my current settings for ublock
[my-ublock-backup_2023-08-10_16.06.01.txt](https://github.com/uBlockOrigin/uBlock-issues/files/12314176/my-ublock-backup_2023-08-10_16.06.01.txt)
 
Incase it's an extension conflict here are the ones I have
![image](https://github.com/uBlockOrigin/uBlock-issues/assets/26977075/fc12000b-c0cc-47af-bb11-cf02d625e95b)

**Reply by stephenhawk8054** [reply]
Turning off the target extension (uBO) is not the correct way to test if it has issue.

The correct way to test is turning off all other extensions and test with your target extension (uBO) only:

- If it causes issue again -> Easier to investigate.

- If it does not, turn on your other extensions one-by-one and see at which point does the issue occur again -> Easier to narrow down the conflictions.

**Reply by Josh65-2201** [reply]
I've tried it with Ublock only and it still happens but not as bad. I will try the one by one for the other extensions to see what makes it worse
Here's the profiler with only Ublock enabled
[Edge freeze Ublock only.json.zip](https://github.com/uBlockOrigin/uBlock-issues/files/12330344/Edge.freeze.Ublock.only.json.zip)

**Reply by gorhill** [reply, marked as answer]
As per this profile, uBO is not your issue, it barely shows up in the profiling data, 49ms/8000ms (0.6%), it's pretty much all duckduckgo code in there, Best is that you report your issue to browser devs if you experience freezes, it's not uBO causing your issue, your profiling data shows this.
